halotools/mock_observables/counts_in_cells/counts_in_cylinders.py

```diff
@@ -191,6 +209,12 @@ def _counts_in_cylinders_process_args(sample1, sample2, proj_search_radius, cyli
     y2 = sample2[:, 1]
     z2 = sample2[:, 2]
 
+    if return_indexes and ((len(x1) > 2e32) or (len(x2) > 2e32)):
```

Comment:
I don't know when sims will start having > 4 billion objects, but this is already memory intensize and I thought it would be much more common to have someone unable to run this because they were running out of memory vs having someone unable to run this because they have too many objects (and boatloads of ram).

Reply:
This algorithm would be unusably slow for samples so large. Present-day sims/mocks can exceed these numbers, and Halotools does not have canned functions that deal with that. What is needed is to distribute (buffered) subvolumes across different compute nodes, which is beyond scope for v0.x. So, fine to either delete this line or leave as is, doesn't really matter.